### ISSUE TITLE
Add per thread monitoring and logging

### ISSUE THREAD

**Issue by SmokinCache** [JerichoJones/JJ-s-XMR-STAK-HashRate-Monitor-and-Restart-Tool#10]
Rather than monitoring overall hash rate monitor it per thread. This would allow great ease in finding the offended GPU causing issues. I have found that once a GPU crashes mining software will continue to function for 30 seconds to hours reporting a n/a has rate in XMR-STAK.. If that n/a status or low hash rate could be written to a log before windows had time to BSOD it has the potential to save days in tuning a multigpu system. My GPU intensities are all equal so a global hash rate drop setting will do, but for those that run different intensity thread on the same GPU I am not sure a global setting would work.

**Comment by Paulius07** [comment]
Yes sometimes one of the card does not reset properly and mining starts with lower overall hash, so fixed value checking will help too, example: if (currHash < 1900 ) - restart. Because after testing cards, we all know what starting hash should be.

**Comment by JerichoJones** [comment]
I need to talk to Fireiceuk about how the thread data works. It does not appears to ID it in the JSON output.
The starting hash thing I'm looking at.

**Comment by JerichoJones** [comment]
Currently, the JSON API for XMR-STAK does not differentiate threads by device. I don't know if that will change. If it does I will see about working it in.

**Comment by ruky87** [comment]
But we could differentiate the threads manually inside the script... If we know what GPU is associated to the index... Could be it possible?

**Comment by SmokinCache** [comment]
Moved from other issue and closed

Device ID by the monitor is not important. I mapped which GPUs are running which thread. This is easy to do watching the hash rate and manipulating performance in ODT.

SLOT,	FANCTRL,TEMP,	FREQ/hash, Thread
1 Vega64 in PCIEX16_1 , OD1	, GPU3,	GPU5, 10/11
2 Vega56 in PCIEX16_3, OD2,	GPU0,	GPU3, 6/7
3 Vega64 in PCIEX1_1, OD3,	GPU2,	GPU2, 4/5
4 Vega56 in PCIEX1_3 , OD4,	GPU5,	GPU1, 2/3
5 Vega56 in PCIEX1_2 , OD5,	GPU4	GPU4, 8/9
6 Vega56 in PCIEX1_4, OD6	, GPU1,	GPU0, 0/1

**Comment by borzaka** [comment]
It would be nice if the logs showed what data was received before resetting.

I often get this:
`2017-12-19_16:59	3.2	Hash Mon Phase: STAK is not returning good data. Current Hash =  `

I assume one of my card is crashed due to overclocking, but I don't know which one. The crashed card shows `(na)` hashrate. The log could contain that thread(s) which showed `(na)`.
